### ISSUE TITLE
2.1.1 broken arguments-based validation

### ISSUE THREAD

**Issue by panta82** [anarchicknight/react-native-communications#31]
You switched from classes with ordinary functions to `export const` arrow  functions.

According to this https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/Arrow_functions:

> An arrow function expression has a shorter syntax compared to function expressions and does not bind its own this, arguments, super, or new.target. 

In your code, you have this:

```javascript
export const phonecall = (phoneNumber, prompt) => {
	if(arguments.length !== 2) {
			console.log('you must supply exactly 2 arguments');
			return;
		}
```

Which gets compiled into this:

```javascript
var phonecall=exports.phonecall=function phonecall(phoneNumber,prompt){
if(_arguments.length!==2){
console.log('you must supply exactly 2 arguments');
return;
}
```

As you can see, arguments gets drawn from the outer scope, instead from your lambda function. This breaks validation.

**Comment by anarchicknight** [comment]
thanks - will hopefully look at this tonight after work

**Comment by anarchicknight** [comment]
should be resolved by 2.1.2 which has just been published
